app/Http/Controllers/InBoundEmailController.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\User;
+use App\InBoundEmail;
+use App\ContactFieldType;
+use Illuminate\Http\Request;
+
+class InBoundEmailController extends Controller
+{
+    /**
+     * Create a new controller instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        $this->middleware('webhook');
+    }
+
+    public function new_postmark(Request $request)
```

Comment:
I'm not convinced to call this method `new_postmark`.

I don't think we treat the problem the right way. The way we associate an email to an account doesn't seem to be super secure.

We can't rely on the from address to find an user account - this can be super easily spoofed and then we'll associate an email to the wrong account.

In my opinion:
1. we should create an unique ID for each user in the database, that we will append to the email address people will use to contact Postmark after the `+` sign, like `dropbox+HASHID@postmark`.
1. Postmark will give us the `hashid` in the `MailboxHash` parameter in the JSON they send back.
1. This will let us identify, with the combination of the `from` email address, the user we need to associate the email with.
1. Then we proceed.

Reply:
The method only works for postmark that needs to be clear. Later we could add new_mailgun and other services. Perhaps new_inbound_postmark?

Good point I agree on the hashing. I would prefer to use the existing hash method it would be easy to expand for users. It might be harder to make it shorter though. Do you think we should aim for a short hash or not care?

Reply:
@turtles2 the problem with the current hash is the fact that it's prefixed with `h:` and I don't know how the `+` sign will work with a `:` after it... so that means we would need to get rid of this `:`. By the way, I think the longer the key after the `+` sign, the better.

Reply:
Also, @asbiin is in the process of adding email verification to Monica. We should make sure that we will only accept (as the `from` address) an email that has been verified already.

Reply:
It has been a bit since I implemented it but I feel like adding a method that meets what we need for the email will be easy I will did into it this weekend hopefully

Reply:
@djaiss You are correct that : dose not work. But a . will work. We could change the preflix or do a replace on the string before giving it to hashIds?